Slack Alerts for failures categorized by BFA plugin

johan1252 opened the pull request:
Commit adds Slack Notifications to the BFA plugin by making use of the Slack Notifier plugin.
By default,
-Slack notifications in BFA are turned off
-The slack channel specified in Slack Notifier will be used (if slack option enabled)
-All failure categories will be reported (if slack option enabled)

The BFA plugin settings now include:
-Toggle for turning on/off slack notifications
-Ability to specify which failure cause categories for which to notify.
-Ability to override the slack channel used for the BFA plugin

Unit tests were added in BuildFailureScannerHudsonTest for the areas of code added to the failure scanning.
As well as tests added to SlackMessageProviderTest for tests specific to the new class created.

With advanced menu not expanded:
![screen shot 2018-08-14 at 4 11 02 pm](https://user-images.githubusercontent.com/6536680/44115756-b2f56d88-9fdc-11e8-853d-b621602e6934.png)

With advanced menu expanded:
![screen shot 2018-08-14 at 4 11 11 pm](https://user-images.githubusercontent.com/6536680/44115755-b2d32516-9fdc-11e8-85b7-4cd3d6bd225f.png)

BenD10 reviewed (approved):
Only suggestion would be removing the changes to format/whitespace, or atleast move them into a seperate commit for easier readability. Otherwise not a big deal.

johan1252 committed:
Commit adds Slack Notifications to the BFA plugin by making use of the Slack Notifier plugin.
By default,
-Slack notifications in BFA are turned off
-The slack channel specified in Slack Notifier will be used (if slack option enabled)
-All failure categories will be reported (if slack option enabled)

The BFA plugin settings now include:
-Toggle for turning on/off slack notifications
-Ability to specify which failure cause categories for which to notify.
-Ability to override the slack channel used for the BFA plugin

Unit tests were added in BuildFailureScannerHudsonTest for the areas of code added to the failure scanning.
As well as tests added to SlackMessageProviderTest for tests specific to the new class created.
